internal/v1/handler_blueprints.go

```diff
@@ -81,30 +75,67 @@ func (bb *BlueprintBody) RedactPasswords() {
 	}
 }
 
-func (u *User) setPassword(password *string) {
-	if password != nil {
-		if u.Password == nil {
-			u.Password = new(string)
-		}
-		*u.Password = *password
+// Merges Password or SshKey from other User struct to this User struct if it is not set
+func (u *User) MergeExisting(other User) {
+	if u.Password == nil {
+		u.Password = other.Password
+	}
+	if u.SshKey == nil {
+		u.SshKey = other.SshKey
 	}
 }
 
-func (u *User) setSshKey(sshKey *string) {
-	if sshKey != nil {
-		if u.SshKey == nil {
-			u.SshKey = new(string)
-		}
-		*u.SshKey = *sshKey
+// User must have name and non-empty password or ssh key
+func (u *User) hasValidValues() error {
+	nameRe := regexp.MustCompile(`\S+`)
+	validName := nameRe.MatchString(u.Name)
+	validPassword := u.Password != nil && len(*u.Password) > 0
+	validSshKey := u.SshKey != nil && len(*u.SshKey) > 0
+	isValid := validName && (validPassword || validSshKey)
+	if !isValid {
```

Comment:
What’s up with this short-lived variables you use everywhere? This could have been simply:

```
if !(validName && (validPassword || validSshKey)) {
```

Reply:
changed